segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeperstorage/LogStorage.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.bookkeeperstorage;
+
+import io.pravega.common.util.ImmutableDate;
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentSkipListMap;
+import javax.annotation.concurrent.GuardedBy;
+import lombok.Getter;
+
+/*
+ * Storage Log represents a single segment. A segment is rolled over whenever a ownership change is observed.
+ *
+ **/
+class LogStorage {
+    @GuardedBy("this")
+    final ConcurrentSkipListMap<Integer, LedgerData> dataMap;
```

Comment:
This is mutable so it should be private.

Reply:
Fixed.